kernel/ixgbe/src/lib.rs

```diff
@@ -1198,10 +1248,49 @@ impl IxgbeNic {
 }
 
 /// Possible link speeds of the 82599 NIC
-enum LinkSpeedMbps {
-    LS100 = 0x1 << 28,
-    LS1000 = 0x2 << 28,
-    LS10000 = 0x3 << 28
+#[derive(PartialEq)]
+pub enum LinkSpeedMbps {
+    LS100 = 100,
+    LS1000 = 1000,
+    LS10000 = 10000, 
+    LSUnknown = 0,
+}
+
+/// Converts between a LinkSpeedMbps enum and a u32 number.
+/// The u32 number is the value in the links register that represents the link speed.
+impl From<u32> for LinkSpeedMbps {
+    fn from(item: u32) -> Self {
+        if item == (0x1 << 28) {
+            Self::LS100
+        } else if item == (0x2 << 28) {
+            Self::LS1000
+        } else if item == (0x3 << 28) {
+            Self::LS10000
+        } else {
+            Self::LSUnknown
+        }
+    }
+}
+
```

Comment:
Generally the `From` trait is the right idea, but unfortunately this documentation is not available to the caller of this because it's a trait implementation. So there's no way to actually view this documentation. Thus, the caller of this type would most likely expect that calling `LinkSpeedMbps::from(100u32)` would give an object `LS100`, when in reality it will give `LSUnknown`. 

As such, instead of using the `From` trait, I think you just ought to just define a direct conversion function that makes it abundantly clear what the u32 value represents, basically what you already wrote above.

For example:
```rust
impl LinkSpeedMbps {
    [pub] fn from_links_register_value(register_value_or_whatever_its_called: u32) -> LinkSpeedMbps {
        match value {
            1 << 28 => Self::LS100,
            2 << 28 => Self::LS1000,
            3 << 28 => Self::LS10000,
             _ => Self::Unknown,
        }
    }
```